### PR TITLE
Use a single build script

### PR DESCRIPTION
**Pull request by jngrad**
The `cuda_build.sh` used to be a separate build script for GPU jobs, but it is now just a two-line wrapper for `build_cmake.sh`. It has been in that state for almost 2 years and keeps getting thinned out, to the point where it would make more sense to merge both files.

### CODE REVIEW

File: maintainer/CI/build_cmake.sh [hunk 1]
```diff
@@ -42,6 +42,8 @@ function end {
 
 # execute and output a command
 # handle environment variables
+[ -z "$cuda_job" ] && cuda_job="false"
+[ $cuda_job = "true" ] && [ -n "${CI_PROJECT_DIR}" ] && srcdir="${CI_PROJECT_DIR}"
```
Comment (fweik) [inline maintainer/CI/build_cmake.sh:46]: Why is this needed?

Comment (jngrad) [reply maintainer/CI/build_cmake.sh:46]: `cuda_build.sh` overrides the user-provided `$srcdir` with `${CI_PROJECT_DIR}`, but if the script is not run in CI, `$srcdir` becomes an empty string, then gets the value `pwd`.
This behavior can be confusing when running the script on a local machine, so I added an extra `[ -n "${CI_PROJECT_DIR}" ]` to avoid overriding `$srcdir` when the script is not run in CI.

Comment (fweik) [reply maintainer/CI/build_cmake.sh:46]: Yes, why does it overwrite srcdir? Isn't that just the default?

Comment (jngrad) [reply maintainer/CI/build_cmake.sh:46]: It probably has to do with the documentation jobs (`deploy_sphinx_documentation`, `check_sphinx`, etc.) which start with the command `cd ${CI_PROJECT_DIR}/build`. We could explicitly pass `srcdir="${CI_PROJECT_DIR}"` to the job from which the documentation jobs depend and drop the overwriting line from the build script.

**Review by fweik — APPROVED** [review]
